cmd/cdi-controller/controller.go

```diff
@@ -372,24 +385,53 @@ func getTokenPrivateKey() *rsa.PrivateKey {
 // Note: objects you read once with the controller runtime client are cached.
 // TODO: Make our watches way more specific using labels, for example,
 // at the point of writing this, we don't care about VolumeSnapshots without the CDI label
-func getNewManagerCache(cdiNamespace string) cache.NewCacheFunc {
+func getCacheOptions(apiClient client.Client, cdiNamespace string) cache.Options {
 	namespaceSelector := fields.Set{"metadata.namespace": cdiNamespace}.AsSelector()
-	return cache.BuilderWithOptions(
-		cache.Options{
-			SelectorsByObject: cache.SelectorsByObject{
-				&networkingv1.Ingress{}: {
-					Field: namespaceSelector,
-				},
-				&routev1.Route{}: {
```

Comment:
Just making sure, we don't want route anymore?

Reply:
We conditionally add `&routev1.Route{}` if running on openshift

Reply:
here's some context - https://github.com/kubernetes-sigs/controller-runtime/issues/2456